src/modules/xine/common.c

```diff
@@ -0,0 +1,180 @@
+	yadif->usrc  = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->uvpitch);
+	yadif->vsrc  = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->uvpitch );
+	yadif->yprev = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->ypitch );
+	yadif->uprev = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->uvpitch );
+	yadif->vprev = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->uvpitch );
+	yadif->ynext = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->ypitch );
+	yadif->unext = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->uvpitch );
+	yadif->vnext = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->uvpitch );
+	yadif->ydest = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->ypitch );
+	yadif->udest = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->uvpitch );
+	yadif->vdest = (unsigned char *) mlt_pool_alloc( yadif->yheight * yadif->uvpitch );
+
+	return yadif;
+}
+
+static void close_yadif(yadif_filter *yadif)
+{
+	mlt_pool_release( yadif->ysrc );
+	mlt_pool_release( yadif->usrc );
+	mlt_pool_release( yadif->vsrc );
+	mlt_pool_release( yadif->yprev );
+	mlt_pool_release( yadif->uprev );
+	mlt_pool_release( yadif->vprev );
+	mlt_pool_release( yadif->ynext );
+	mlt_pool_release( yadif->unext );
+	mlt_pool_release( yadif->vnext );
+	mlt_pool_release( yadif->ydest );
+	mlt_pool_release( yadif->udest );
+	mlt_pool_release( yadif->vdest );
+	mlt_pool_release( yadif );
+
+#if defined(__GNUC__) && !defined(PIC)
+	// Set SSSE3 bit to cpu
+	asm (\
+	"mov $1, %%eax \n\t"\
+	"push %%ebx \n\t"\
+	"cpuid \n\t"\
+	"pop %%ebx \n\t"\
+	"mov %%ecx, %%edx \n\t"\
+	"shr $9, %%edx \n\t"\
+	"and $1, %%edx \n\t"\
+	"shl $9, %%edx \n\t"\
+	"and $511, %%ebx \n\t"\
+	"or %%edx, %%ebx \n\t"\
+	: "=b"(yadif->cpu) : "p"(yadif->cpu) : "%eax", "%ecx", "%edx");
+#endif
+}
+
+mlt_deinterlacer supported_method( mlt_deinterlacer method )
+{
+	// Round the method down to the next one that is supported
+	if ( method <= mlt_deinterlacer_onefield ) method = mlt_deinterlacer_onefield;
+	else if ( method <= mlt_deinterlacer_linearblend ) method = mlt_deinterlacer_linearblend;
+	else if ( method <= mlt_deinterlacer_weave ) method = mlt_deinterlacer_weave;
+	else if ( method <= mlt_deinterlacer_bob ) method = mlt_deinterlacer_bob;
+	else if ( method <= mlt_deinterlacer_greedy ) method = mlt_deinterlacer_greedy;
+	else if ( method <= mlt_deinterlacer_yadif_nospatial ) method = mlt_deinterlacer_yadif_nospatial;
+	else if ( method <= mlt_deinterlacer_invalid ) method = mlt_deinterlacer_yadif;
+	return method;
+}
+
+int deinterlace_image( mlt_image dst, mlt_image src, mlt_image prev, mlt_image next, int tff, mlt_deinterlacer method )
+{
+	if ( !dst || !src || !dst->data || !src->data )
+	{
+		return 1;
+	}
+
+	if ( method >= mlt_deinterlacer_yadif_nospatial &&
+		 ( !prev || !next || !prev->data || !next->data ) )
+	{
+		method = mlt_deinterlacer_linearblend;
```

Comment:
It looks like this is the same behavior as the existing filter, but I think now upon review while testing onefield is a better looking fallback as its appearance is much closer to what yadif produces.

Reply:
I had thought the same thing. But I didn't care to break from the tradition since I am replacing it with the ffmpeg based deinterlacers. I suppose it only occurs on the first frame (when there is no previous) and the last frame (when there is no next). That will not be an issue with FFMpeg based deinterlacers.

Reply:
Another way to deal with missing frames is to repeat the correct frame.